src/python/WMCore/MicroService/MSUnmerged/MSUnmergedRSE.py

```diff
@@ -145,3 +157,22 @@ def writeRSEToMongoDB(self, collection, fullRSEToDB=False, overwrite=False):
             return True
         else:
             return False
+
+    def resetRSE(self, collection, keepTimestamps=False, keepCounters=False):
+        """
+        Resets all records of the RSE object to default values  and write the
+        document to MongoDB
+        :param keepTimestamps: Bool flag to keep the timestamps
+        :param keepCounters:   Bool flag to keep the counters
+        :return:               True if operation succeeded.
+        """
+        resetDoc = self.defaultDoc()
+
+        if keepTimestamps:
+            resetDoc['timestamps'] = self['timestamps']
+        if keepCounters:
+            resetDoc['counters'] = self['counters']
+
+        self.update(resetDoc)
+        writeResult = self.writeRSEToMongoDB(collection, fullRSEToDB=True, overwrite=True)
```

Comment:
Isn't `fullRSEToDB` meant to be parameterized? It's not clear to me why it's set to True here.

Reply:
And I have just found a bug which caused the the transformation of the data type for some fields when we writing/resetting the RSE record to MongoDB, e.g. 
```
In [67]: rse.resetRSE(msUnmergedColl)
Out[67]: True

In [68]: rse
Out[68]: 
{'name': 'T2_DE_DESY',
 'pfnPrefix': None,
 'isClean': False,
 'timestamps': {'rseConsStatTime': 0.0,
  'prevStartTime': 0.0,
  'startTime': 0.0,
  'prevEndTime': 0.0,
  'endTime': 0.0},
 'counters': {'totalNumFiles': 0,
  'dirsToDeleteAll': 0,
  'dirsToDelete': 0,
  'filesToDelete': 0,
  'deletedSuccess': 0,
  'deletedFail': 0},
 'files': {'allUnmerged': [],
  'toDelete': {},
  'protected': {},
  'deletedSuccess': [],
  'deletedFail': []},
 'dirs': {'allUnmerged': set(), 'toDelete': set(), 'protected': set()},
 '_id': ObjectId('61a0e052b713a7692b8475f9')}

In [69]: msUnmergedColl.find_one(rse.mongoFilter)
Out[69]: 
{'_id': ObjectId('61a0e052b713a7692b8475f9'),
 'name': 'T2_DE_DESY',
 'pfnPrefix': None,
 'isClean': False,
 'timestamps': {'rseConsStatTime': 0.0,
  'prevStartTime': 0.0,
  'startTime': 0.0,
  'prevEndTime': 0.0,
  'endTime': 0.0},
 'counters': {'totalNumFiles': 0,
  'dirsToDeleteAll': 0,
  'dirsToDelete': 0,
  'filesToDelete': 0,
  'deletedSuccess': 0,
  'deletedFail': 0},
 'files': {'allUnmerged': [],
  'toDelete': [],
  'protected': [],
  'deletedSuccess': [],
  'deletedFail': []},
 'dirs': {'allUnmerged': [], 'toDelete': [], 'protected': []}}
```

Notice the type of all the fields which are of filter/generator type (e.g. `rse['files']['toDelete']`). The fix is in place already, but this one was tricky to debug and I was just lucky to notice it before merging the PR.